No try..catch tambem temos o finally para finalizar a tarefa quando der erro, assim voce nao fica travado em uma tela.
